src/System.ServiceModel.UnixDomainSocket/src/System/ServiceModel/UnixDomainSocketTransportSecurity.cs

```diff
@@ -0,0 +1,182 @@
+        }
+
+        [DefaultValue(DefaultProtectionLevel)]
+        public ProtectionLevel ProtectionLevel
+        {
+            get { return _protectionLevel; }
+            set
+            {
+                if (!Security.ProtectionLevelHelper.IsDefined(value))
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentOutOfRangeException(nameof(value)));
+                }
+
+                _protectionLevel = value;
+            }
+        }
+
+        public ExtendedProtectionPolicy ExtendedProtectionPolicy
+        {
+            get
+            {
+                return _extendedProtectionPolicy;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(value));
+                }
+
+                if (value.PolicyEnforcement == PolicyEnforcement.Always &&
+                    !ExtendedProtectionPolicy.OSSupportsExtendedProtection)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
+                        new PlatformNotSupportedException(SR.ExtendedProtectionNotSupported));
+                }
+                _extendedProtectionPolicy = value;
+            }
+        }
+
+        [DefaultValue(UnixDomainSocketTransportDefaults.SslProtocols)]
+        public SslProtocols SslProtocols
+        {
+            get { return _sslProtocols; }
+            set
+            {
+                SslProtocolsHelper.Validate(value);
+                _sslProtocols = value;
+            }
+        }
+
+        private SslStreamSecurityBindingElement CreateSslBindingElement(bool requireClientCertificate)
+        {
+            if (_protectionLevel != ProtectionLevel.EncryptAndSign)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(
+                    SR.UnsupportedSslProtectionLevel, _protectionLevel)));
+            }
+
+            SslStreamSecurityBindingElement result = new SslStreamSecurityBindingElement();
+            result.RequireClientCertificate = requireClientCertificate;
+            result.SslProtocols = _sslProtocols;
+            return result;
+        }
+
+        internal BindingElement CreatePosixIdentityOnlyBinding()
+        {
+            return new UnixPosixIdentityBindingElement();
+        }
+
+        private static bool IsSslBindingElement(BindingElement element, UnixDomainSocketTransportSecurity transportSecurity)
+        {
+            SslStreamSecurityBindingElement ssl = element as SslStreamSecurityBindingElement;
+            if (ssl == null)
+            {
+                return false;
+            }
+
+            transportSecurity.ProtectionLevel = ProtectionLevel.EncryptAndSign;
+            return true;
+        }
+
+        internal BindingElement CreateTransportProtectionOnly()
+        {
```

Comment:
This method should be used when SecurityMode is Transport and ClientCredentialType is None. We still need to secure the connection use TLS, we just don't provide a client certificate.

Reply:
https://github.com/CoreWCF/CoreWCF/issues/1249